tsdb/head.go

```diff
@@ -857,13 +858,15 @@ func (h *Head) Truncate(mint int64) (err error) {
 		return ok
 	}
 	h.metrics.checkpointCreationTotal.Inc()
-	if _, err = wal.Checkpoint(h.logger, h.wal, first, last, keep, mint); err != nil {
+	cpStats, err := wal.Checkpoint(h.logger, h.wal, first, last, keep, mint)
+	if err != nil {
 		h.metrics.checkpointCreationFail.Inc()
 		if _, ok := errors.Cause(err).(*wal.CorruptionErr); ok {
 			h.metrics.walCorruptionsTotal.Inc()
 		}
 		return errors.Wrap(err, "create checkpoint")
 	}
+	h.lastCheckpointName = cpStats.Name
```

Comment:
If Prometheus restarts, won't you miss the existing checkpoint?

Reply:
Addressed.